### PR TITLE
docs: Add SafeArea docs

### PR DESCRIPTION
**Pull request by kazo0**

### CODE REVIEW

File: doc/controls/SafeArea.md [hunk 1]
```diff
@@ -0,0 +1,181 @@
+# SafeArea
+
+`SafeArea` is a specialized control that overrides the `Padding` or `Margin` properties of its child/attached control to ensure that its inner content is always within the [`ApplicationView.VisibleBounds`](https://docs.microsoft.com/en-us/uwp/api/windows.ui.viewmanagement.applicationview.visiblebounds?view=winrt-22621) rectangle.
+
+The `ApplicationView.VisibleBounds` is the rectangular area of the screen which is completely unobscured by any window decoration, such as the status bar, rounded screen corners or any type of screen notch.
```
Comment (agneszitte-nventive) [inline doc/controls/SafeArea.md:5]:
```suggestion
The `ApplicationView.VisibleBounds` is the rectangular area of the screen which is completely unobscured by any window decoration, such as the status bar, rounded screen corners, or any type of screen notch.
```

File: doc/controls/SafeArea.md [hunk 2]
```diff
@@ -0,0 +1,181 @@
...
+
+The `SafeArea` can also be used to ensure that the specified control will never be hidden by any sort of soft-input panel, such as the on-screen keyboard. This is done by observing the state of the keyboard and treating the area that it occupies when open as part of the "unsafe" area of the screen.
```
Comment (agneszitte-nventive) [inline doc/controls/SafeArea.md:7]:
```suggestion
The `SafeArea` can also be used to ensure that the specified control will never be hidden by any sort of soft-input panels, such as the on-screen keyboard. This is done by observing the state of the keyboard and treating the area that it occupies when open as part of the "unsafe" area of the screen.
```

File: doc/controls/SafeArea.md [hunk 3]
```diff
@@ -0,0 +1,181 @@
...
+
+### Using `InsetMode.Padding` versus `InsetMode.Margin`
+
+The default `Mode` for `SafeArea` is set to `InsetMode.Padding`. Using the Padding property as the `SafeArea` inset ensures that your control's content will never be obscured by the "unsafe" area but still allows things like the control's Background color to "bleed" into the unsafe area. To highlight this feature, refer to the example below.
+
+#### Example
+Here we are using a the Toolkit's [`TabBar`](TabBarAndTabBarItem.md) with both the `TopTabBarStyle` and the `BottomTabBarStyle`. Both controls have their `Background`s set to `Purple`, note the differences within the unsafe areas of the screen between the `Padding` mode and the `Margin` mode.
```
Comment (jeromelaban) [inline doc/controls/SafeArea.md:62]:
```suggestion
Here we are using the Toolkit's [`TabBar`](TabBarAndTabBarItem.md) with both the `TopTabBarStyle` and the `BottomTabBarStyle`. Both controls have their `Background`s set to `Purple`. Note the differences within the unsafe areas of the screen between the `Padding` mode and the `Margin` mode.
```

File: doc/controls/SafeArea.md [hunk 4]
```diff
@@ -0,0 +1,181 @@
...
+
+```diff
+<utu:TabBar Grid.Row="2"
++           utu:SafeArea.Insets="Bottom"
+            Background="Purple">
+```
+
+![safearea_with_padding_alpha](../assets/safearea_with_padding.png)
+
+# [**Margin**](#tab/margin)
```
Comment (jeromelaban) [inline doc/controls/SafeArea.md:156]: You'll need to finish the last section with a `---`

**Review by jeromelaban — APPROVED** [review]

**Review by agneszitte-nventive — APPROVED** [review]

**Pull request merged**